### ISSUE TITLE
Neos should remember to start in seated mode

### ISSUE THREAD

**Issue by Enkiko** [Frooxius/NeosPublic#939]
Most people I know use Neos in seated mode. At the moment that means they have to set Neos to seated mode every time they log on. This is unnecessary use of time, so I suggest an option to remember the choice of seated mode, so that Neos starts in seated mode.

**Comment by H3BO3** [comment]
If it's any help figuring out the logic for stand detection - I propose that if the groundprojectedheight moves into and stays within a certain threshold of the set user height for a sustained period of time (perhaps a few seconds?), seated mode switches off.

This would be evaluated on startup too. If the conditions are not met the toggle and the last "set" seated mode height persists.

**Comment by CuriousNicholi** [comment]
If this goes through, please make it optional because my work is calibrated to start in specific positions, and spawning into seated mode would knock all my work off balance.

**Comment by H3BO3** [comment]
@CuriousNicholi Do you even use seated mode, in that case? It would only persist based on how it was set previously. If it was already off there would be no change on startup.

**Comment by mralext20** [comment]
Previously from froox: 
> That's a bit problematic. Technically it's not a problem for it to remember your seated mode offset, but it might be wrong next time you launch. E.g. you turn on seated mode and then exit Neos.
> 
> Say next time you start Neos you are standing up, now you're floating above the floor because it thinks you're still sitting. 
> 
> Similarly if it turns on when you start up Neos, it can't assume that you're sitting properly. Maybe you have your headset on the floor when you start Neos and you end up with wrong seated position. Or maybe you're standing up and it won't do much.
> 
> One thing I'm thinking of doing is auto-disabling seated position when you end up too far above your standing position as a result, that might fix enough issues and allow it to be remembered.


https://discordapp.com/channels/402159838827905024/439422161657135106/743628085303181412

**Comment by H3BO3** [comment]
@mralext20 Hence I suggested logic for maintaining the GroundProjectedHeight from the previous session, which fixes the problems addressed in paras 1 and 3, and also logic for disabling it while standing, fixing the problems in paras 2 and 4.

**Comment by Hayden-da-Serg** [comment]
If this logic is implemented then I suggest the sensitivity of it be up to the user, since certain people may not like a default setting, for instance I joke around all the time with the fact that I can set seated mode then ascend into the air while t-posing, other people may also want this setting so they can configure their range for what they most feel comfortable with.

**Comment by H3BO3** [comment]
@Hayden-da-Serg It would likely be an internal threshold that compares with the set user height that is already in the settings. Staying within ~6 inches or so (or higher) would be a fairly reliable indicator that someone is standing up. To be honest, I don't think joke behavior is a relevant factor here. I suggest the AvatarTrackingOffset component instead.

**Comment by Frooxius** [comment]
Yes, I plan on adding this soon! My plan was to add pretty much the mechanism for auto-disabling seated mode as described, where if you're above threshold for too long, it auto toggles off the seated mode. 

One possibility is for only soft disable on that one and automatically re-enabling it when you sit back down. That way if people are standing up, they can use Neos as normal and then sit down and have the seated mode still active.

The only thing with that is that it might be a bit confusing if they try to pick something up or crouch and end up pulled back up. It's a question if that'd be less or more confusing than suddenly realizing that their seated mode is off without turning it off (although I do want to add a notification message when this happens which should help there). 

I can add a setting for remembering the mode, but for fine tuned configuration you'd have to wait until reworked Settings.

**Comment by RileyGuy** [comment]
If neos could also remember free-form dash, that would be splendid as well.
